### PR TITLE
#337/fit edit chat details in avatar page template

### PR DESCRIPTION
**Pull request by Damrod**

### CODE REVIEW

**Comment by Damrod** [comment]
Here I had a lot to adjust, because whenever I fixed something, something else was breaking. This is the end result.

**Comment by Damrod** [comment]
There is some problem with the placeholder avatars (WILDCARD_AVATAR_URL), but they are transitory anyway.

File: webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx [hunk 1]
```diff
@@ -23,6 +28,20 @@ type AvatarPageTemplateProps = {
...
+
+type AvatarPageCaptionLikeTemplateProps = AvatarPageTemplateCommonProps & {
+  variant: AvatarPageVariants;
+  captionLikeElement: JSX.Element;
+};
+
+type AvatarPageTemplateProps =
+  | AvatarPageRegularTemplateProps
+  | AvatarPageCaptionLikeTemplateProps;
+
```
Comment (gbudau) [inline webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx:44]: suggestion: What do you think about adding the prop `captionLikeElement?: JSX.Element;` to the `AvatarPageTemplateProps`. This way, we could remove the variant and do something like this:

```tsx
          <div className="avatar-page-caption-like-wrapper">
            {captionLikeElement && captionLikeElement}
          </div>
```

File: webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx [hunk 2]
```diff
@@ -78,64 +73,54 @@ function JoinChatroom({ chatroomId }: JoinChatroomProps) {
...
   return (
-    <div className="join-chatroom">
-      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
-        {`channel/${chatroom.name}`}
-      </Header>
-      <div className="join-chatroom-avatar">
-        <LargeAvatar
-          url={
+    <div className="join-chatroom-page">
+      <AvatarPageTemplate
```
Comment (gbudau) [inline webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx:78]: praise: Thanks for adding the template to these pages.

**Review by gbudau — APPROVED** [review]
Thank you for these improvements. LGTM :rocket:

File: webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx (hunk 1 above)
Comment (Damrod) [reply webapp/src/shared/components/templates/AvatarPageTemplate/AvatarPageTemplate.tsx]: done. Thx!

File: webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx (hunk 2 above)
Comment (Damrod) [reply webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx:78]: Thank you!

**Pull request merged**